src/main/scala/com/imaginea/activegrid/core/models/NodeReport.scala

```diff
@@ -8,3 +12,52 @@ case class NodeReport(override val id: Option[Long],
                       response: String,
                       taskReports: List[TaskReport],
                       status: StepExecutionStatus) extends BaseEntity
+
+object NodeReport {
+  val labelName = "NodeReport"
+  val nodeReportAndInstance = "HAS_Instance"
+  val nodeReportAndTaskReport = "HAS_TaskReport"
+  val logger = Logger(LoggerFactory.getLogger(getClass.getName))
+
+  implicit class NodeReportImpl(nodeReport: NodeReport) extends Neo4jRep[NodeReport] {
+    override def toNeo4jGraph(entity: NodeReport): Node = {
+      val map = Map("response" -> entity.response,
+        "status" -> entity.status.stepExecutionStatus)
+      val parentNode = Neo4jRepository.saveEntity(labelName, entity.id, map)
+      val instanceNode = entity.node.toNeo4jGraph(entity.node)
+      Neo4jRepository.createRelation(nodeReportAndInstance, parentNode, instanceNode)
+      entity.taskReports.foreach { taskReport =>
+        val childNode = taskReport.toNeo4jGraph(taskReport)
+        Neo4jRepository.createRelation(nodeReportAndTaskReport, parentNode, childNode)
+      }
+      parentNode
+    }
+
+    override def fromNeo4jGraph(id: Long): Option[NodeReport] = {
+      NodeReport.fromNeo4jGraph(id)
+    }
+  }
+
+  def fromNeo4jGraph(id: Long): Option[NodeReport] = {
+    val mayBeNode = Neo4jRepository.findNodeById(id)
+    mayBeNode.map { node =>
+      val map = Neo4jRepository.getProperties(node, "response", "status")
+      val instanceNodeId = Neo4jRepository.getChildNodeId(id, nodeReportAndInstance)
+      val mayBeInstance = instanceNodeId.flatMap(nodeId => Instance.fromNeo4jGraph(nodeId))
+      val instance = mayBeInstance match {
```

Comment:
If we use for comprehension then it will return Option[Instance], in that case field(Instance) should be optional. But here Instance is mandatory field and we need to throw exception.